src/ci/notifier.rs

```diff
@@ -0,0 +1,10 @@
+pub mod github;
+pub mod gitlab;
+use crate::template;
+
+use anyhow::Result;
+
+pub trait Notifiable {
+    fn notify(&self, body: template::Template, patch: bool) -> Result<()>;
+    fn job_url(&self) -> &String;
```

Comment:
[nits]
```rust
fn job_url(&self) -> String;
```

I prefer `String` instead of `&String` because of portability of `job_url`.
The following should be also updated

```rust
   fn job_url(&self) -> String {
        &self.job_url.clone()
    }
```

Reply:
@hiro-o918 Thanks for the review! fixed in [5bedd5b](https://github.com/hirosassa/ksnotify/pull/50/commits/5bedd5b0b02f28c5329e244d476bc15ac8b2fd5c)